### ISSUE TITLE
 `keras.ops.repeat` cannot return an exptected shape when `x` is a `KerasTensor` and the `axis` is `None`

### ISSUE THREAD

**Issue by Zhaopudark** [keras-team/keras#19821]
Hello. Thank you for your contributions and maintenance for the best Keras.

I'm following the instructions of [Conditional GAN (code samples, uses Keras 3)](https://keras.io/examples/generative/conditional_gan/), and focusing on the `keras.ops.repeat` function that is used in it.

I have found, maybe, if the input tensor of  `keras.ops.repeat`  is a symbolic tensor, i.e., the `keras.KerasTensor`, and the arg `axis` is `None`, the returned one will not be my expected one.

As the following：
```python
batch_size = 64
class_num = 10
a = keras.KerasTensor(shape=(batch_size, class_num), dtype=tf.float32)
a = a[:, :, None, None] # [B,10,1,1]
b = keras.ops.repeat(a, repeats=[28 * 28])
print(b.shape)# (784,)
# expected output: (501760,)
```

If assign `axis`, it works as expected:
```python
a = keras.KerasTensor(shape=(batch_size, class_num), dtype=tf.float32)
a = a[:, :, None, None] # [B,10,1,1]
b = keras.ops.repeat(a, repeats=[28 * 28],axis=0)
print(b.shape)# (784, 10, 1, 1)
# expected output: (784, 10, 1, 1)
```

If not use the symbolic tensor, it also works as expected:
```python
a = keras.random.normal(shape=(batch_size, class_num), dtype=tf.float32)
a = a[:, :, None, None] # [B,10,1,1]
b = keras.ops.repeat(a, repeats=[28 * 28])
print(b.shape)# (501760,)
# expected output: (501760,)
```

So, is the above a bug? 

And my environment is:
- Keras: Version: 3.3.3
- Numpy: Version: 1.26.4
- TensorFlow: Version: 2.16.1

Thanks in advance.

**Comment by pandyah5** [comment]
I can look into this and report my findings in a few hours

**Comment by Grvzard** [comment]
This is due to an oversight caused by the different ways Keras and other backends handle the `repeats` parameter.
You can submit a PR after you solve it.

**Comment by pandyah5** [comment]
Hi @Grvzard thank you for your inputs. I was able to reproduce the behaviour on my end but I do have a follow up. Based on the documentation of [tf.keras.ops.repeat](https://www.tensorflow.org/api_docs/python/tf/keras/ops/repeat) the input tensor if flattened by default before the repeat is applied to it.

Based on OP's example, the [64, 10, 1, 1] tensor will be flattened to [640,]. This is a scalar as we expect. Hence when we repeat a scalar we get an output tensor shape of [784,]. Isn't this the expected behavior? Please correct me if I am missing something.

I verified that this is what is happening by the following test code:

```
batch_size = 64
class_num = 10
a = keras.KerasTensor(shape=(batch_size, class_num), dtype=tf.float32)
a = a[:, :, None, None] # [B,C,1,1]
b = keras.ops.repeat(a, repeats=[28 * 28])
print(b.shape)

c = keras.KerasTensor(shape=(640,), dtype=tf.float32)
c = keras.ops.repeat(c, repeats=[28 * 28])
print(c.shape)
```
Output:
```
(784,)
(784,)
```
